pkg/bindings/test/containers_test.go

```diff
@@ -0,0 +1,192 @@
+		Expect(data.State.Status).To(Equal("running"))
+	})
+
+	It("podman remove a paused container by id without force", func() {
+		// Removing a paused container without force should fail
+		var name = "top"
+		bt.RunTopContainer(&name, &falseFlag, nil)
+		data, err := containers.Inspect(connText, name, nil)
+		Expect(err).To(BeNil())
+		err = containers.Pause(connText, data.ID)
+		Expect(err).To(BeNil())
+		err = containers.Remove(connText, data.ID, &falseFlag, &falseFlag)
+		Expect(err).ToNot(BeNil())
+		code, _ := bindings.CheckResponseCode(err)
+		Expect(code).To(BeNumerically("==", http.StatusInternalServerError))
+	})
+
+	It("podman remove a paused container by id with force", func() {
+		// Removing a paused container with force should work
+		var name = "top"
+		bt.RunTopContainer(&name, &falseFlag, nil)
+		data, err := containers.Inspect(connText, name, nil)
+		Expect(err).To(BeNil())
+		err = containers.Pause(connText, data.ID)
+		Expect(err).To(BeNil())
+		err = containers.Remove(connText, data.ID, &trueFlag, &falseFlag)
+		Expect(err).To(BeNil())
+	})
+
+	It("podman stop a paused container by name", func() {
+		// Stopping a paused container by name should fail
+		var name = "top"
+		bt.RunTopContainer(&name, &falseFlag, nil)
+		err := containers.Pause(connText, name)
+		Expect(err).To(BeNil())
+		err = containers.Stop(connText, name, nil)
+		Expect(err).ToNot(BeNil())
+		code, _ := bindings.CheckResponseCode(err)
+		Expect(code).To(BeNumerically("==", http.StatusInternalServerError))
+	})
+
+	It("podman stop a paused container by id", func() {
+		// Stopping a paused container by id should fail
+		var name = "top"
+		bt.RunTopContainer(&name, &falseFlag, nil)
+		data, err := containers.Inspect(connText, name, nil)
+		Expect(err).To(BeNil())
+		err = containers.Pause(connText, data.ID)
+		Expect(err).To(BeNil())
+		err = containers.Stop(connText, data.ID, nil)
+		Expect(err).ToNot(BeNil())
+		code, _ := bindings.CheckResponseCode(err)
+		Expect(code).To(BeNumerically("==", http.StatusInternalServerError))
+	})
+
+	It("podman stop a running container by name", func() {
+		// Stopping a running container by name should work
+		var name = "top"
+		bt.RunTopContainer(&name, &falseFlag, nil)
+		err := containers.Stop(connText, name, nil)
+		Expect(err).To(BeNil())
+
+		// Ensure container is stopped
+		data, err := containers.Inspect(connText, name, nil)
+		Expect(err).To(BeNil())
+		Expect(data.State.Status).To(Equal("exited"))
+	})
+
+	It("podman stop a running container by ID)", func() {
+		// Stopping a running container by ID should work
+		var name = "top"
+		bt.RunTopContainer(&name, &falseFlag, nil)
+		data, err := containers.Inspect(connText, name, nil)
+		Expect(err).To(BeNil())
+		err = containers.Stop(connText, data.ID, nil)
+		Expect(err).To(BeNil())
+
+		// Ensure container is stopped
+		data, err = containers.Inspect(connText, name, nil)
+		Expect(err).To(BeNil())
+		Expect(data.State.Status).To(Equal("exited"))
+	})
+
+})
```

Comment:
What happens when we try to pause a paused container? Same with pause a stopped container? And same for unpause.

Reply:
ack, I'll add tests for those after lunch